### PR TITLE
Refactor listeners

### PR DESCRIPTION
**Pull request by k88hudson**
- Adds constants for addon-to-content, content-to-addon
- Consolidates listeners for addon's communication with content
- Consolidate listeners for places change events

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-stream/791)

### CODE REVIEW

**Comment by k88hudson** [comment]
Refactor of https://github.com/mozilla/activity-stream/pull/761

**Comment by k88hudson** [comment]
@oyiptong R? also @rlr since you looked at this before?

File: lib/ActivityStreams.js [hunk 1]
```diff
@@ -238,75 +246,56 @@ ActivityStreams.prototype = {
...
-    this.off(am.type("NOTIFY_BLOCK_URL"), this._respondToPlacesRequests);
-    this.off(am.type("NOTIFY_UNBLOCK_URL"), this._respondToPlacesRequests);
-    this.off(am.type("NOTIFY_UNBLOCK_ALL"), this._respondToPlacesRequests);
-    this.off(am.type("NOTIFY_BOOKMARK_DELETE"), this._respondToPlacesRequests);
-    this.off(am.type("SEARCH_STATE_REQUEST"), this._respondToSearchRequests);
-    this.off(am.type("NOTIFY_PERFORM_SEARCH"), this._respondToSearchRequests);
-    this.off(am.type("NOTIFY_ROUTE_CHANGE"), this._onRouteChange);
-    this.off(am.type("NOTIFY_USER_EVENT"), this._handleUserEvent);
-    this.off(am.type("EXPERIMENTS_REQUEST"), this._respondToExperimentsRequest);
+    this.off(CONTENT_TO_ADDON, this._handlers);
```
Comment (rlr) [inline lib/ActivityStreams.js]: I think you meant `this._contentToAddonHandlers` here ^^

**Comment by rlr** [comment]
This is much cleaner IMO. I likey. I'll defer to @oyiptong

File: test/test-ActivityStreams-message-passing.js [hunk 2]
```diff
@@ -2,74 +2,101 @@
...
 const test = require("sdk/test");
 const tabs = require("sdk/tabs");
+const {before, after} = require("sdk/test/utils");
 const {ActivityStreams} = require("lib/ActivityStreams");
 const httpd = require("./lib/httpd");
 const {doGetFile} = require("./lib/utils");
+const {CONTENT_TO_ADDON} = require("common/event-constants");
 
 const PORT = 8099;
+let path = "/dummy-activitystreams.html";
```
Comment (oyiptong) [inline test/test-ActivityStreams-message-passing.js]: please use `kVarName` or `VAR_NAME` for globals

**Comment by oyiptong** [comment]
r+ with a nit and @rlr's comment.

Much cleaner!

File: lib/ActivityStreams.js (hunk 1 above)
Comment (k88hudson) [inline lib/ActivityStreams.js]: oops! good catch

**Pull request merged**